web_transport/sdk/impl/web_transport_server_backend.h

```diff
@@ -40,8 +40,8 @@ class WebTransportServerBackend : public WebTransportSessionVisitor {
 
  private:
   WebTransportServerInterface::Visitor* visitor_;
-  std::unordered_map<::quic::WebTransportSessionId,
-                     std::unique_ptr<WebTransportServerSession>>
+  // Key is HTTP/3 QUIC stream connection ID.
+  std::unordered_map<std::string, std::unique_ptr<WebTransportServerSession>>
```

Comment:
make sure you guard access to this map.

Reply:
It's only called when headers are received, which is expected to be called on the same thread. So no lock is checked here.

Reply:
you will need this eventually as you'll remove ids from the map.

Reply:
For what I can imagine at this time, an ID is removed when a session is terminated by either sending a FIN, or receiving a FIN. Both of them are expected to be called on IO thread, which is the same thread for receiving headers. A thread checker is added to ensure session map is accessed on the same thread.